### PR TITLE
Illusions of dist-ribution

### PR DESCRIPTION
**Pull request by Dhaulagiri**
## :pushpin: Summary

Make all component tests work while maintaining the illusion that the published version of tokens is the same as what we are working with on disk.

## :hammer_and_wrench: Detailed description

I'm not convinced this is "good", but it does work. This PR does two main things.

First, it creates symlinks in the root of the `packages/tokens` folder pointing to the corresponding folder in `dist`. This allows us to reference assets in that folder similar to how they are in the published version of the package which is from the root *not* `dist/`. This effectively restores the code to what it was before the monorepo migration.

Second, when it comes time to publish we remove these symlinks so we can actually copy the folders to root for publishing. This is necessary because `npm publish` [does not follow symlinks](https://github.com/npm/npm/issues/3310). Once we are done publishing, the symlinks are manually restored via a separate script.

So, this does work but it feels a bit overcomplicated when we could publish these files from `dist/` and avoid the need for all of this.

```bash
~/c/w/d/p/tokens (br-hoist|✚2) $ ls -l
total 96
-rw-r--r--  1 runnells  staff   3060 Mar 11 09:54 CONTRIBUTING.md
-rw-r--r--  1 runnells  staff  16725 Mar 11 09:54 LICENSE.md
-rw-r--r--  1 runnells  staff   1381 Mar 11 09:54 README.md
-rw-r--r--  1 runnells  staff   2499 Mar 11 09:54 RELEASE.md
lrwxr-xr-x  1 runnells  staff     11 Mar 15 08:31 devdot@ -> dist/devdot
drwxr-xr-x  5 runnells  staff    160 Mar 11 15:27 dist/
lrwxr-xr-x  1 runnells  staff      9 Mar 15 08:31 docs@ -> dist/docs
drwxr-xr-x  7 runnells  staff    224 Mar 15 08:23 node_modules/
-rw-r--r--  1 runnells  staff   1190 Mar 15 08:24 package.json
lrwxr-xr-x  1 runnells  staff     13 Mar 15 08:31 products@ -> dist/products
drwxr-xr-x  7 runnells  staff    224 Mar 15 08:24 scripts/
drwxr-xr-x  5 runnells  staff    160 Mar 11 09:54 src/
-rw-r--r--  1 runnells  staff  10885 Mar 11 09:54 tsconfig.json
~/c/w/d/p/tokens (br-hoist|✚2) $ npm publish --dry-run

> @hashicorp/design-system-tokens@0.5.1 prepublishOnly
> ./scripts/publish.sh

npm notice 
npm notice 📦  @hashicorp/design-system-tokens@0.5.1
npm notice === Tarball Contents === 
npm notice 16.7kB  LICENSE.md                         
npm notice 1.4kB   README.md                          
npm notice 3.7kB   devdot/css/helpers/color.css       
npm notice 1.1kB   devdot/css/helpers/elevation.css   
npm notice 158B    devdot/css/helpers/focus-ring.css  
npm notice 2.6kB   devdot/css/helpers/typography.css  
npm notice 21.0kB  devdot/css/tokens.css              
npm notice 133.1kB docs/devdot/tokens.json            
npm notice 132.3kB docs/products/tokens.json          
npm notice 1.2kB   package.json                       
npm notice 3.6kB   products/css/helpers/color.css     
npm notice 1.1kB   products/css/helpers/elevation.css 
npm notice 158B    products/css/helpers/focus-ring.css
npm notice 2.6kB   products/css/helpers/typography.css
npm notice 20.8kB  products/css/tokens.css            
npm notice === Tarball Details === 
npm notice name:          @hashicorp/design-system-tokens          
npm notice version:       0.5.1                                    
npm notice filename:      @hashicorp/design-system-tokens-0.5.1.tgz
npm notice package size:  26.0 kB                                  
npm notice unpacked size: 341.3 kB                                 
npm notice shasum:        cd984b038be77bc3784ec08080243c9b98cfe3f4 
npm notice integrity:     sha512-glSqbJXPWC8Nk[...]oyB52Wg/A59TA== 
npm notice total files:   15                                       
npm notice 

> @hashicorp/design-system-tokens@0.5.1 postpublish
> ./scripts/postPublish.sh

+ @hashicorp/design-system-tokens@0.5.1
~/c/w/d/p/tokens (br-hoist|✔) $ ls -l
total 96
-rw-r--r--  1 runnells  staff   3060 Mar 11 09:54 CONTRIBUTING.md
-rw-r--r--  1 runnells  staff  16725 Mar 11 09:54 LICENSE.md
-rw-r--r--  1 runnells  staff   1381 Mar 11 09:54 README.md
-rw-r--r--  1 runnells  staff   2499 Mar 11 09:54 RELEASE.md
lrwxr-xr-x  1 runnells  staff     11 Mar 15 08:37 devdot@ -> dist/devdot
drwxr-xr-x  5 runnells  staff    160 Mar 11 15:27 dist/
lrwxr-xr-x  1 runnells  staff      9 Mar 15 08:37 docs@ -> dist/docs
drwxr-xr-x  7 runnells  staff    224 Mar 15 08:23 node_modules/
-rw-r--r--  1 runnells  staff   1190 Mar 15 08:24 package.json
lrwxr-xr-x  1 runnells  staff     13 Mar 15 08:37 products@ -> dist/products
drwxr-xr-x  7 runnells  staff    224 Mar 15 08:24 scripts/
drwxr-xr-x  5 runnells  staff    160 Mar 11 09:54 src/
-rw-r--r--  1 runnells  staff  10885 Mar 11 09:54 tsconfig.json
```

## :white_check_mark: Reviewer's checklist

- [x] +1 Percy if applicable
- [x] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

**Comment by amyrlam** [comment]
Nice - this change seems a-ok, but please let me test a few things on my machine, will report back

//

IMO removing the `/dist` from .npmignore and publishing dist or other could be a more maintainable long-term solution as a follow-on
- Right now, I think it's confusing as `.npmignore` is usually more like a `.gitignore`... vs excluding a whole directory
- if `/dist` is the issue, we can have the files output to another directory? I don't 100% follow the `/dist` reasoning yet but IIRC @didoo is not tied to keeping `/dist` it was more about tidyness?

File: .github/workflows/ci-components.yml [hunk 1]
```diff
@@ -56,34 +56,34 @@ jobs:
...
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: yarn
+          cache-dependency-path: yarn.lock
+      - name: Install Dependencies
+        run: yarn install --frozen-lockfile
+      - name: Run Tests
+        working-directory: packages/components
+        run: ./node_modules/.bin/ember try:one ${{ matrix.try-scenario }}
```
[the remark this answers is not in the document]
Comment (amyrlam) [reply .github/workflows/ci-components.yml:89]: 👍 works for me

**Review by amyrlam — APPROVED** [review]
thanks :shipit: 

//

left some comments if want to do a follow-on that may be easier for others to debug, but no strong opinion

ember-canary passes locally, if we want to turn that back on in a follow-on

**Comment by Dhaulagiri** [comment]
> IMO removing the /dist from .npmignore and publishing dist or other could be a more maintainable long-term solution as a follow-on

I agree and after discussing with @didoo  there seems to be alignment across the team for this change. I can work up a PR for it when I get a minute. It might be worth landing this PR as a stopgap until we can get a new version of the package published, but I'll figure that out when I do the other PR.

**Comment by Dhaulagiri** [comment]
I'll bring this in. I have two follow-ups that should unwind us from this more permanently.

**Pull request merged**